mne/viz/_3d.py

```diff
@@ -593,6 +598,16 @@ def plot_alignment(info=None, trans=None, subject=None, subjects_dir=None,
 
     if isinstance(surfaces, str):
         surfaces = [surfaces]
+    if isinstance(surfaces, dict):
+        user_alpha = surfaces
+        # check that alpha values contain floats between 0 and 1 or None
+        check_values = [x is None or (isinstance(x, (float, int)) and x >= 0
+                                      and x <= 1) for x in user_alpha.values()]
+        if not all(check_values):
+            raise ValueError("If surfaces is dict, its values must contain "
+                             "floats between 0 and 1 or None")
```

Comment:
```suggestion
        for key, value in user_alpha.items():
            _validate_type(key, str, f'surfaces key {repr(key)}')
            _validate_type(value, (None, 'numeric'), f'surfaces[{repr(key)}]'
            if value is not None:
                user_alpha[key] = float(value)
                if not 0 <= user_alpha[key] <= 1:
                    raise ValueError(
                        f'surfaces[{repr(key)}] ({value}) must be between 0 and 1')
```
This will be more specific and give TypeError or ValueError as appropriate